pypeit/flatfield.py

```diff
@@ -487,6 +509,17 @@ def build_mask(self):
                 bpmflats[bpm] = self.slits.bitmask.turn_on(bpmflats[bpm], flag)
         return bpmflats
 
+    def build_waveimg(self):
```

Comment:
We execute these steps in several places and are replicating that code evyrehere, i.e. extract, build_wavecalib etc. Should we have a core utility routine somewhere that executes them?